Automattic/wp-calypso
# Domains: Update new domain management list component with logic from the original one

#### Changes proposed in this Pull Request

There's a lot of business logic code in the original `domain-management/List` component, so let's re-add that to the new component. That way, we can be more sure that the logic and exceptions remain the same.

#### Testing instructions

Open the live Calypso link, go to "Upgrades > domains" and append this feature flag to your browser's URL: `?flags=domains/management-list-redesign`.

Access different sites with different domain configurations: sites with one or more registered or mapped domains, sites with unused domain credit, etc. Ensure the site domains list page looks and works fine.
